### PR TITLE
Feature/issue 1057 partial localization

### PR DESCRIPTION
**Pull request by tomy-pg**

### CODE REVIEW

File: src/components/services/service_list_location_search_component.tsx [hunk 1]
```diff
@@ -261,8 +264,12 @@ const getSearchOnPress = (
     }
 };
 
+const locationInputIsMyLocation = (locationInput: string, i18n: I18n): boolean => (
+    locationInput === MY_LOCATION || isLocalizedMyLocation(locationInput, i18n)
+);
```
Comment (rasmus-storjohann-PG) [inline src/components/services/service_list_location_search_component.tsx]: You have three functions that do almost the same thing ...

File: src/api/fetch_lat_long_from_location.ts [hunk 2]
```diff
@@ -4,28 +4,40 @@ import { LatLong } from '../validation/latlong/types';
...
+    location === MY_LOCATION
+);
+
+const locationIsNotEmpty = (location: string): boolean => (
+    location !== ''
+);
+
+const userIsOnline = (onlineStatus: OnlineStatus): boolean => (
+    onlineStatus === OnlineStatus.Online
+);
```
Comment (rasmus-storjohann-PG) [inline src/api/fetch_lat_long_from_location.ts]: Nice

File: src/components/services/service_list_location_search_component.tsx [hunk 3]
```diff
@@ -239,78 +238,24 @@ const LocateButton = (props: LocateButtonProps): JSX.Element => (
...
-    const latLong = await fetchFn().catch(handleFetchError);
-    setIsFetchingLatLng(false);
+    const latLong = await fetchLatLongFromLocation(location);
     setManualUserLocation({
-        label: locationLabel,
-        latLong,
+        label: locationInputValue,
+        latLong: latLong,
     });
+    setIsFetchingLatLng(false);
```
Comment (rasmus-storjohann-PG) [inline src/components/services/service_list_location_search_component.tsx:274]: Looking for improved error handling later in this PR...

File: src/components/services/service_list_location_search_component.tsx [hunk 4]
```diff
@@ -146,6 +139,31 @@ const ExpandedSearch = (props: ExpandedSearchProps): JSX.Element => {
...
+): JSX.Element => {
+    if (!hasMyLocation(locationInputValue, i18n)) {
+        return (
+            <LocateButton
+            onPress={getLocateOnPress(setLocationInputValue, i18n)}
+            isFetchingLatLng={isFetchingLatLng}
+            />
+        );
+    }
+    return <EmptyComponent />;
```
Comment (rasmus-storjohann-PG) [inline src/components/services/service_list_location_search_component.tsx:156]: Picky: I'd flip this if statement around, two benefits: one less negation, and early return on the "don't need to do anything" branch.

File: src/components/services/service_list_location_search_component.tsx (hunk 3 above)
Comment (rasmus-storjohann-PG) [reply src/components/services/service_list_location_search_component.tsx:274]: So the only thing I would really ask you to fix in this PR is here. Before you call `fetchLatLongFromLocation()` you call `setIsFetchingLatLng()` with true, and after it comes back you set it back to false. But what if `fetchLatLongFromLocation()` throws, then the flag never gets set back. What we need here is some form of a `finally` control flow. Can you make that work? Other than that, the other comments are optional and this thing is ready to go.

Comment (tomy-pg) [reply src/components/services/service_list_location_search_component.tsx:274]: Fixed.

File: src/components/services/service_list_location_search_component.tsx (hunk 4 above)
Comment (tomy-pg) [reply src/components/services/service_list_location_search_component.tsx:156]: Fixed. I think this makes it easier to understand as well.

**Pull request merged**